### PR TITLE
Support named return arguments in native_functions.

### PR DESCRIPTION
**Pull request by gchanan**
Note there was a hacky way of doing this before by specifying "return:" lists manually; this makes the
return names part of the function declaration itself.

### CODE REVIEW

File: aten/src/ATen/native/README.md [hunk 1]
```diff
@@ -126,6 +112,37 @@ Here are the supported default values:
...
+
+Tuple return:
+```
+(ReturnType [retarg0], ReturnType [retarg1], ...)
+```
+
+The following are permissible on ReturnType:
+- `Tensor` and `TensorList`, which translate into the C++ types `Tensor` and `std::vector<Tensor>`,
+  respectively (unless the operation is in-place, in which case the return type
+  is `Tensor&`.
```
Comment (ezyang) [inline aten/src/ATen/native/README.md]: Oh nooooo, can we not do this please. The fact that we return a mutable tensor ref is a mistake, let's not enshrine it in the syntax of native functions. (Didn't we have a proposal for syntax for mutation?)

File: aten/src/ATen/native_parse.py [hunk 2]
```diff
@@ -99,6 +103,33 @@ def parse_arguments(args, func_decl, func_name, func_return):
     return arguments
 
 
+def parse_return_arguments(return_decl, inplace):
```
Comment (ezyang) [inline aten/src/ATen/native_parse.py]: What changes were made in this block?

Comment (gchanan) [reply aten/src/ATen/native_parse.py]: this is totally new, it just borrows some minimal code from the normal argument parsing.

File: aten/src/ATen/native/README.md (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/native/README.md]: This isn't relevant; there are a couple of things going on here:
1) This section is all literally moved, besides the part that explains you can now give return arguments optional names.  I just moved it below all the argument examples for clarity.
2) The part you are objecting to (which again, is literally moved), is referring to the C++ return type, not the native function return type.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**